src/main/java/io/jsonwebtoken/JwtParser.java

```diff
@@ -338,6 +358,60 @@
     <T> T parse(String jwt, JwtHandler<T> handler)
             throws ExpiredJwtException, UnsupportedJwtException, MalformedJwtException, SignatureException, IllegalArgumentException;
 
+    
+    /**
+     * Parses the header part of the specified compact serialized JWT string.
+     * 
+     * <p>Note that while parsing header no validation or security check will be applied.</p>
+     * 
+     * @param jwt the compact serialized JWT to parse
+     * @return Deserialized {@link Header} part of specified compact JWT string.
+     */
+	Header parseHeader(String jwt);
+
+	/**
+	 * Checks if the specified compact serialized JWT string is expired based on the builder's current configuration state. 
+     * <p>
+     * <p>Note that you need this method probably only then, if the parser is configured to by pass 'expiration time' ({@code exp}) claim validation (by calling builder's {@link #ignoreExpiry()} method).</p>
+     *
+     * @param jwt the compact serialized JWT to check
+     * @return {@code true} if the specified JWT compact string is expired, {@code false} otherwise.
+	 */
+	boolean isExpired(String jwt);
```

Comment:
Do you mean isPremature or isExpired or both? In our use case we need really first ignore the expiration validation and then separately check if the token expired or not. I don't really know, if there are other cases for isPremature out there. From my point of view, we can kick all ignoreNotBefore, isPremature and validateNotBefore methods entirely. But why not let them there, may be some one need them for a specific use case.